internal/debug/debug_debug.go

```diff
@@ -0,0 +1,48 @@
+//go:build debug
+// +build debug
+
+package debug
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+)
+
+var (
+	debugLevel = 1
+	logger     = log.New(os.Stderr, "", log.Lmicroseconds)
+)
+
+func init() {
+	level, err := strconv.Atoi(os.Getenv("DEBUG_LEVEL"))
+	if err != nil {
+		return
+	}
+
+	debugLevel = level
+}
+
+// Log writes the formatted string to stderr.
+// Level indicates the verbosity of the messages to log.
+// The greater the value, the more verbose messages will be logged.
+func Log(level int, format string, v ...interface{}) {
```

Comment:
Doesn't have to be done now, but it'd be good to document the expected range. I think we use 1 and 3, so maybe those should get constants and names.

Reply:
My preference would be to remove it entirely as I don't think it provides much value.  I think we can do better with our logs in general, I opened https://github.com/Azure/go-amqp/issues/187 to track this.